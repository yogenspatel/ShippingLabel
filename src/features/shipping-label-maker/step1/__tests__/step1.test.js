/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import ShippingLabelStep1 from "../index";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Step 1 Component renders correctly", () => {
  it("has rendered without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ShippingLabelStep1 />, div);
  });
  it("has called snapshot with ShippingLabel component", () => {
    const tree = shallow(<ShippingLabelStep1 />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("has called componentDidMount method", () => {
    const spy = jest.spyOn(ShippingLabelStep1.prototype, 'componentDidMount');
    const props = {
        setShippingData: {},
        getShippingData: jest.fn()
    }
    const wrapper = shallow(<ShippingLabelStep1 {...props} />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
    expect(props.getShippingData).toBeCalled();
    spy.mockClear();
  });
  it("has retrieved data on componentDidMount method", () => {
    const shippingData = {
        to: {
            name: "test 1 to",
            street: "test street 1 to",
            city: "test city 1 to",
            state: "test state 1 to",
            zip: "test zip 44444 to"
        }
    }
    const spy = jest.spyOn(ShippingLabelStep1.prototype, 'componentDidMount');
    const props = {
        setShippingData: shippingData,
        getShippingData: jest.fn()
    }
    const wrapper = shallow(<ShippingLabelStep1 {...props} />);
    wrapper.instance().componentDidMount();
    expect(wrapper.state('name')).toEqual(shippingData.to.name);
    expect(wrapper.state('street')).toEqual(shippingData.to.street);
    expect(wrapper.state('city')).toEqual(shippingData.to.city);
    expect(wrapper.state('state')).toEqual(shippingData.to.state);
    expect(wrapper.state('zip')).toEqual(shippingData.to.zip);
    
    spy.mockClear();
  });
});
