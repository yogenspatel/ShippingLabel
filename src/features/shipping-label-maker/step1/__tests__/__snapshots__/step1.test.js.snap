// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step 1 Component renders correctly has called snapshot with ShippingLabel component 1`] = `
<div
  className="card"
>
  <div
    className="h4 card-header"
  >
    Enter the Sender‘s Address
  </div>
  <div
    className="card-body"
  >
    <RenderFormField
      context={
        ShippingLabelStep1 {
          "context": Object {},
          "key": "to",
          "props": Object {
            "getShippingData": [Function],
            "setShippingData": Object {},
          },
          "refs": Object {},
          "renderFormFields": [Function],
          "setState": [Function],
          "state": Object {
            "city": "",
            "errorObj": Object {
              "city": true,
              "name": true,
              "state": true,
              "street": true,
              "zip": true,
            },
            "name": "",
            "state": "",
            "street": "",
            "zip": "",
          },
          "updater": Updater {
            "_callbacks": Array [],
            "_renderer": ReactShallowRenderer {
              "_context": Object {},
              "_didScheduleRenderPhaseUpdate": false,
              "_dispatcher": Object {
                "readContext": [Function],
                "useCallback": [Function],
                "useContext": [Function],
                "useDebugValue": [Function],
                "useEffect": [Function],
                "useImperativeHandle": [Function],
                "useLayoutEffect": [Function],
                "useMemo": [Function],
                "useReducer": [Function],
                "useRef": [Function],
                "useState": [Function],
              },
              "_element": <ShippingLabelStep1
                getShippingData={[Function]}
                setShippingData={Object {}}
              />,
              "_firstWorkInProgressHook": null,
              "_forcedUpdate": false,
              "_instance": [Circular],
              "_isReRender": false,
              "_newState": null,
              "_numberOfReRenders": 0,
              "_renderPhaseUpdates": null,
              "_rendered": <div
                className="card"
              >
                <div
                  className="h4 card-header"
                >
                  Enter the Sender‘s Address
                </div>
                <div
                  className="card-body"
                >
                  <RenderFormField
                    context={[Circular]}
                    fieldName="name"
                    id="sender_0"
                    onChange={[Function]}
                    placeHolder="Enter name"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="street"
                    id="sender_1"
                    onChange={[Function]}
                    placeHolder="Enter street"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="city"
                    id="sender_2"
                    onChange={[Function]}
                    placeHolder="Enter city"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="state"
                    id="sender_3"
                    onChange={[Function]}
                    placeHolder="Enter state"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="zip"
                    id="sender_4"
                    onChange={[Function]}
                    placeHolder="Enter zip"
                    type="text"
                  />
                </div>
              </div>,
              "_rendering": false,
              "_updater": [Circular],
              "_workInProgressHook": null,
            },
          },
        }
      }
      fieldName="name"
      id="sender_0"
      key="sender_0"
      onChange={[Function]}
      placeHolder="Enter name"
      type="text"
    />
    <RenderFormField
      context={
        ShippingLabelStep1 {
          "context": Object {},
          "key": "to",
          "props": Object {
            "getShippingData": [Function],
            "setShippingData": Object {},
          },
          "refs": Object {},
          "renderFormFields": [Function],
          "setState": [Function],
          "state": Object {
            "city": "",
            "errorObj": Object {
              "city": true,
              "name": true,
              "state": true,
              "street": true,
              "zip": true,
            },
            "name": "",
            "state": "",
            "street": "",
            "zip": "",
          },
          "updater": Updater {
            "_callbacks": Array [],
            "_renderer": ReactShallowRenderer {
              "_context": Object {},
              "_didScheduleRenderPhaseUpdate": false,
              "_dispatcher": Object {
                "readContext": [Function],
                "useCallback": [Function],
                "useContext": [Function],
                "useDebugValue": [Function],
                "useEffect": [Function],
                "useImperativeHandle": [Function],
                "useLayoutEffect": [Function],
                "useMemo": [Function],
                "useReducer": [Function],
                "useRef": [Function],
                "useState": [Function],
              },
              "_element": <ShippingLabelStep1
                getShippingData={[Function]}
                setShippingData={Object {}}
              />,
              "_firstWorkInProgressHook": null,
              "_forcedUpdate": false,
              "_instance": [Circular],
              "_isReRender": false,
              "_newState": null,
              "_numberOfReRenders": 0,
              "_renderPhaseUpdates": null,
              "_rendered": <div
                className="card"
              >
                <div
                  className="h4 card-header"
                >
                  Enter the Sender‘s Address
                </div>
                <div
                  className="card-body"
                >
                  <RenderFormField
                    context={[Circular]}
                    fieldName="name"
                    id="sender_0"
                    onChange={[Function]}
                    placeHolder="Enter name"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="street"
                    id="sender_1"
                    onChange={[Function]}
                    placeHolder="Enter street"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="city"
                    id="sender_2"
                    onChange={[Function]}
                    placeHolder="Enter city"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="state"
                    id="sender_3"
                    onChange={[Function]}
                    placeHolder="Enter state"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="zip"
                    id="sender_4"
                    onChange={[Function]}
                    placeHolder="Enter zip"
                    type="text"
                  />
                </div>
              </div>,
              "_rendering": false,
              "_updater": [Circular],
              "_workInProgressHook": null,
            },
          },
        }
      }
      fieldName="street"
      id="sender_1"
      key="sender_1"
      onChange={[Function]}
      placeHolder="Enter street"
      type="text"
    />
    <RenderFormField
      context={
        ShippingLabelStep1 {
          "context": Object {},
          "key": "to",
          "props": Object {
            "getShippingData": [Function],
            "setShippingData": Object {},
          },
          "refs": Object {},
          "renderFormFields": [Function],
          "setState": [Function],
          "state": Object {
            "city": "",
            "errorObj": Object {
              "city": true,
              "name": true,
              "state": true,
              "street": true,
              "zip": true,
            },
            "name": "",
            "state": "",
            "street": "",
            "zip": "",
          },
          "updater": Updater {
            "_callbacks": Array [],
            "_renderer": ReactShallowRenderer {
              "_context": Object {},
              "_didScheduleRenderPhaseUpdate": false,
              "_dispatcher": Object {
                "readContext": [Function],
                "useCallback": [Function],
                "useContext": [Function],
                "useDebugValue": [Function],
                "useEffect": [Function],
                "useImperativeHandle": [Function],
                "useLayoutEffect": [Function],
                "useMemo": [Function],
                "useReducer": [Function],
                "useRef": [Function],
                "useState": [Function],
              },
              "_element": <ShippingLabelStep1
                getShippingData={[Function]}
                setShippingData={Object {}}
              />,
              "_firstWorkInProgressHook": null,
              "_forcedUpdate": false,
              "_instance": [Circular],
              "_isReRender": false,
              "_newState": null,
              "_numberOfReRenders": 0,
              "_renderPhaseUpdates": null,
              "_rendered": <div
                className="card"
              >
                <div
                  className="h4 card-header"
                >
                  Enter the Sender‘s Address
                </div>
                <div
                  className="card-body"
                >
                  <RenderFormField
                    context={[Circular]}
                    fieldName="name"
                    id="sender_0"
                    onChange={[Function]}
                    placeHolder="Enter name"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="street"
                    id="sender_1"
                    onChange={[Function]}
                    placeHolder="Enter street"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="city"
                    id="sender_2"
                    onChange={[Function]}
                    placeHolder="Enter city"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="state"
                    id="sender_3"
                    onChange={[Function]}
                    placeHolder="Enter state"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="zip"
                    id="sender_4"
                    onChange={[Function]}
                    placeHolder="Enter zip"
                    type="text"
                  />
                </div>
              </div>,
              "_rendering": false,
              "_updater": [Circular],
              "_workInProgressHook": null,
            },
          },
        }
      }
      fieldName="city"
      id="sender_2"
      key="sender_2"
      onChange={[Function]}
      placeHolder="Enter city"
      type="text"
    />
    <RenderFormField
      context={
        ShippingLabelStep1 {
          "context": Object {},
          "key": "to",
          "props": Object {
            "getShippingData": [Function],
            "setShippingData": Object {},
          },
          "refs": Object {},
          "renderFormFields": [Function],
          "setState": [Function],
          "state": Object {
            "city": "",
            "errorObj": Object {
              "city": true,
              "name": true,
              "state": true,
              "street": true,
              "zip": true,
            },
            "name": "",
            "state": "",
            "street": "",
            "zip": "",
          },
          "updater": Updater {
            "_callbacks": Array [],
            "_renderer": ReactShallowRenderer {
              "_context": Object {},
              "_didScheduleRenderPhaseUpdate": false,
              "_dispatcher": Object {
                "readContext": [Function],
                "useCallback": [Function],
                "useContext": [Function],
                "useDebugValue": [Function],
                "useEffect": [Function],
                "useImperativeHandle": [Function],
                "useLayoutEffect": [Function],
                "useMemo": [Function],
                "useReducer": [Function],
                "useRef": [Function],
                "useState": [Function],
              },
              "_element": <ShippingLabelStep1
                getShippingData={[Function]}
                setShippingData={Object {}}
              />,
              "_firstWorkInProgressHook": null,
              "_forcedUpdate": false,
              "_instance": [Circular],
              "_isReRender": false,
              "_newState": null,
              "_numberOfReRenders": 0,
              "_renderPhaseUpdates": null,
              "_rendered": <div
                className="card"
              >
                <div
                  className="h4 card-header"
                >
                  Enter the Sender‘s Address
                </div>
                <div
                  className="card-body"
                >
                  <RenderFormField
                    context={[Circular]}
                    fieldName="name"
                    id="sender_0"
                    onChange={[Function]}
                    placeHolder="Enter name"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="street"
                    id="sender_1"
                    onChange={[Function]}
                    placeHolder="Enter street"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="city"
                    id="sender_2"
                    onChange={[Function]}
                    placeHolder="Enter city"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="state"
                    id="sender_3"
                    onChange={[Function]}
                    placeHolder="Enter state"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="zip"
                    id="sender_4"
                    onChange={[Function]}
                    placeHolder="Enter zip"
                    type="text"
                  />
                </div>
              </div>,
              "_rendering": false,
              "_updater": [Circular],
              "_workInProgressHook": null,
            },
          },
        }
      }
      fieldName="state"
      id="sender_3"
      key="sender_3"
      onChange={[Function]}
      placeHolder="Enter state"
      type="text"
    />
    <RenderFormField
      context={
        ShippingLabelStep1 {
          "context": Object {},
          "key": "to",
          "props": Object {
            "getShippingData": [Function],
            "setShippingData": Object {},
          },
          "refs": Object {},
          "renderFormFields": [Function],
          "setState": [Function],
          "state": Object {
            "city": "",
            "errorObj": Object {
              "city": true,
              "name": true,
              "state": true,
              "street": true,
              "zip": true,
            },
            "name": "",
            "state": "",
            "street": "",
            "zip": "",
          },
          "updater": Updater {
            "_callbacks": Array [],
            "_renderer": ReactShallowRenderer {
              "_context": Object {},
              "_didScheduleRenderPhaseUpdate": false,
              "_dispatcher": Object {
                "readContext": [Function],
                "useCallback": [Function],
                "useContext": [Function],
                "useDebugValue": [Function],
                "useEffect": [Function],
                "useImperativeHandle": [Function],
                "useLayoutEffect": [Function],
                "useMemo": [Function],
                "useReducer": [Function],
                "useRef": [Function],
                "useState": [Function],
              },
              "_element": <ShippingLabelStep1
                getShippingData={[Function]}
                setShippingData={Object {}}
              />,
              "_firstWorkInProgressHook": null,
              "_forcedUpdate": false,
              "_instance": [Circular],
              "_isReRender": false,
              "_newState": null,
              "_numberOfReRenders": 0,
              "_renderPhaseUpdates": null,
              "_rendered": <div
                className="card"
              >
                <div
                  className="h4 card-header"
                >
                  Enter the Sender‘s Address
                </div>
                <div
                  className="card-body"
                >
                  <RenderFormField
                    context={[Circular]}
                    fieldName="name"
                    id="sender_0"
                    onChange={[Function]}
                    placeHolder="Enter name"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="street"
                    id="sender_1"
                    onChange={[Function]}
                    placeHolder="Enter street"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="city"
                    id="sender_2"
                    onChange={[Function]}
                    placeHolder="Enter city"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="state"
                    id="sender_3"
                    onChange={[Function]}
                    placeHolder="Enter state"
                    type="text"
                  />
                  <RenderFormField
                    context={[Circular]}
                    fieldName="zip"
                    id="sender_4"
                    onChange={[Function]}
                    placeHolder="Enter zip"
                    type="text"
                  />
                </div>
              </div>,
              "_rendering": false,
              "_updater": [Circular],
              "_workInProgressHook": null,
            },
          },
        }
      }
      fieldName="zip"
      id="sender_4"
      key="sender_4"
      onChange={[Function]}
      placeHolder="Enter zip"
      type="text"
    />
  </div>
</div>
`;
